app/javascript/app/document-capture/hooks/use-i18n.js

```diff
@@ -1,10 +1,65 @@
-import { useContext } from 'react';
+import { createElement, cloneElement, useContext } from 'react';
 import I18nContext from '../context/i18n';
 
+/** @typedef {import('react').FC|import('react').ComponentClass} Component */
+
+/**
+ * Given an HTML string and an object of tag names to React component, returns a new of React node
+ * where the mapped tag names are replaced by the resulting element of the rendered component.
+ *
+ * Note that this is a very simplistic interpolation of HTML. It only supports well-balanced, non-
+ * nested tag names, where there are no attributes or excess whitespace within the tag names. While
+ * the subject markup itself cannot contain attributes, the return value of the component can be any
+ * valid React element, with or without additional attributes.
+ *
+ * @example
+ * ```
+ * formatHTML('Hello <lg-sparkles>world</lg-sparkles>!', {
+ *   'lg-sparkles': ({children}) => <span className="lg-sparkles">{children}</span>
+ * });
+ * ```
+ *
+ * @param {string}                   html     HTML to format.
+ * @param {Record<string,Component>} handlers Mapping of tag names to components.
+ *
+ * @return {import('react').ReactNode}
+ */
+export function formatHTML(html, handlers) {
+  const pattern = new RegExp(`</?(?:${Object.keys(handlers).join('|')})>`, 'g');
```

Comment:
This should probably do something to the effect of [`Object.keys(handlers).map(escapeRegExp)`](https://lodash.com/docs/4.17.15#escapeRegExp), but... rather than decide to pull in Lodash or reimplement its function, I'm fine to let this be a case of "Don't do that" limitation. I'll make a note to add it to the documentation.

Reply:
> This should probably do something to the effect of [`Object.keys(handlers).map(escapeRegExp)`](https://lodash.com/docs/4.17.15#escapeRegExp), but... rather than decide to pull in Lodash or reimplement its function, I'm fine to let this be a case of "Don't do that" limitation. I'll make a note to add it to the documentation.

Added note in 56f2d0b507769526c014cfc982b0d38ae10afef9.